packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -512,13 +514,19 @@ export class Carousel
         break;
       case "Home":
         event.preventDefault();
+        if (this.selectedIndex === 0) {
+          return;
+        }
         this.direction = "backward";
         this.setSelectedItem(0, true);
         break;
       case "End":
         event.preventDefault();
+        if (this.selectedIndex === lastItem) {
```

Comment:
should this move before event.preventDefault so event isn't prevented if nothing happens?